Repo: Integration tests failing on main due to upstream eslint-stylistic changes

### Suggestion

From https://github.com/typescript-eslint/typescript-eslint/actions/runs/8157268910/job/22296673696:

```plaintext
FAIL  tests/flat-config-types.test.ts (32.88 s)
  ● flat-config-types › typescript › should work successfully
    expect(received).toMatchInlineSnapshot(snapshot)
    Snapshot name: `flat-config-types typescript should work successfully 1`
    - Snapshot  - 12
    + Received  +  0
    - node_modules/@stylistic/eslint-plugin-plus/dts/index.d.ts(7,46): error TS2694: Namespace '"/<tmp_folder>/node_modules/@types/eslint/index".ESLint' has no exported member 'RuleModule'.
    - node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
    -   Named property ''comma-dangle'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
    - node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
    -   Named property ''keyword-spacing'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
    - node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
    -   Named property ''lines-around-comment'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
    - node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
    -   Named property ''lines-between-class-members'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
    - node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
    -   Named property ''padding-line-between-statements'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
    -

      19 |
      20 |     // The stylistic type errors: https://github.com/eslint-stylistic/eslint-stylistic/issues/276
    > 21 |     expect(lines).toMatchInlineSnapshot(`
         |                   ^
Error:       22 |       "node_modules/@stylistic/eslint-plugin-plus/dts/index.d.ts(7,46): error TS26[94](https://github.com/typescript-eslint/typescript-eslint/actions/runs/8157268910/job/22296673696#step:5:95): Namespace '"/<tmp_folder>/node_modules/@types/eslint/index".ESLint' has no exported member 'RuleModule'.
Error:       23 |       node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
      24 |         Named property ''comma-dangle'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.

      at toMatchInlineSnapshot (tests/flat-config-types.test.ts:21:19)
      at assertOutput (tools/integration-test-base.ts:1[99](https://github.com/typescript-eslint/typescript-eslint/actions/runs/8157268910/job/22296673696#step:5:100):7)
      at Object.<anonymous> (tools/integration-test-base.ts:120:9)
 › 1 snapshot failed.
```

--- a/packages/integration-tests/tests/flat-config-types.test.ts
+++ b/packages/integration-tests/tests/flat-config-types.test.ts
@@ -7,31 +7,15 @@ typescriptIntegrationTest(
   __filename,
   ['--allowJs', '--esModuleInterop', 'eslint.config.js'],
   out => {
-    const lines = out
-      .split('\n')
-      .filter(
-        line =>
-          // error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.
-          // this is fine for us to ignore in this context
-          !line.includes('error TS18028'),
-      )
-      .join('\n');
+    const lines = out.split('\n').filter(
+      line =>
+        // error TS18028: Private identifiers are only available when targeting ECMAScript 2015 and higher.
+        // this is fine for us to ignore in this context
+        line && !line.includes('error TS18028'),
+    );
 
-    // The stylistic type errors: https://github.com/eslint-stylistic/eslint-stylistic/issues/276
-    expect(lines).toMatchInlineSnapshot(`
-      "node_modules/@stylistic/eslint-plugin-plus/dts/index.d.ts(7,46): error TS2694: Namespace '"/<tmp_folder>/node_modules/@types/eslint/index".ESLint' has no exported member 'RuleModule'.
-      node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
-        Named property ''comma-dangle'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
-      node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
-        Named property ''keyword-spacing'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
-      node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
-        Named property ''lines-around-comment'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
-      node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
-        Named property ''lines-between-class-members'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
-      node_modules/@stylistic/eslint-plugin/dist/dts/rule-options.d.ts(6,11): error TS2320: Interface 'UnprefixedRuleOptions' cannot simultaneously extend types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions'.
-        Named property ''padding-line-between-statements'' of types 'UnprefixedRuleOptions' and 'UnprefixedRuleOptions' are not identical.
-      "
-    `);
+    // The types should not error (e.g. https://github.com/eslint-stylistic/eslint-stylistic/issues/276)
+    expect(lines).toHaveLength(0);
   },
 );
 eslintIntegrationTest(__filename, 'eslint.config.js', true);

